CONTRIBUTING.md

```diff
@@ -0,0 +1,151 @@
+
+Anyone interested in Mypy may review your code.  One of the Mypy core
+developers will merge your pull request when they think it's ready.
+For every pull request, we aim to promptly either merge it or say why
+it's not yet ready; if you go a few days without a reply, please feel
+free to ping the thread with a new comment.
+
+At present the core developers are (alphabetically):
+* David Fisher (@ddfisher)
+* Jukka Lehtosalo (@JukkaL)
+* Greg Price (@gnprice)
+* Guido van Rossum (@gvanrossum)
+
+
+Preparing Changes
+-----------------
+
+Before you begin: if your change will be a significant amount of work
+to write, we highly recommend starting by opening an issue laying out
+what you want to do.  (This is good advice for all kinds of
+open-source projects in general.)  That lets a conversation happen
+early in case other contributors disagree with what you'd like to do
+or have ideas that will help you do it.
+
+The best pull requests are focused, clearly describe what they're for
+and why they're correct, and contain tests for whatever changes they
+make to the code's behavior.  As a bonus these are easiest for someone
+to review, which helps your pull request get merged quickly!  Standard
+advice about good pull requests for open-source projects applies; we
+have [our own writeup](http://www.mypy-lang.org/wiki/GoodPullRequest)
+of this advice.
+
+See also our [coding conventions](http://www.mypy-lang.org/wiki/CodeConventions) --
+which consist mainly of a reference to
+[PEP 8](https://www.python.org/dev/peps/pep-0008/) -- for the code you
+put in the pull request.
+
+You may also find other pages in the
+[Mypy developer guide](http://www.mypy-lang.org/wiki/DeveloperGuides)
+helpful in developing your change.
+
+
+Issue-tracker conventions
+-------------------------
+
+We aim to reply to all new issues promptly.  We'll assign a
+"milestone" to help us track which issues we intend to get to when,
+and may apply "labels" to carry some other information.  Here's what
+our milestones and labels mean.
+
+### Milestones
+
+We use GitHub "milestones" (see [our
+milestones](https://github.com/python/mypy/milestones)) to roughly
+order what we want to do soon and less soon.
+
+This means they represent a combination of priority and scale of work.
+Bugs that aren't a huge deal but do matter to users and don't seem
+like a lot of work to fix generally go in a near milestone; things
+that will take longer may go further out.
+
+Specifically:
+
+* **Numbered milestones** correspond to releases.  These assignments
+  are changeable and issues may be moved earlier or later.
+  Assignments to further-out milestones are especially likely to
+  change.
+* Point releases, like 0.x.y when we're already at 0.x.z, generally
+  have issues that are less work to tackle and whose user-facing
+  impact is small or a bugfix.  Meatier or more radical issues
+  generally go to a full "minor" release, like 0.x.0.
```

Comment:
Maybe? Last I looked carefully at semver I didn't find the specifics of it totally persuasive. And I'm not sure that most projects that claim "we follow semver" actually do follow the specifics of it -- it can be hard to tell -- and [as the semver writeup itself says](http://semver.org/#why-use-semantic-versioning), the whole point of giving it a brand name and a spec full of RFC 2119 capitalized modal phrases is to nail down the specifics.

That view isn't based on a lot of data, though, and I'm certainly open to being persuaded that we should explicitly adopt semver.